airflow/utils/logging.py

```diff
@@ -40,6 +40,16 @@ def logger(self):
             self._logger = logging.root.getChild(self.__class__.__module__ + '.' + self.__class__.__name__)
             return self._logger
 
+    def set_logger_contexts(self, task_instance):
+        """
+        Set the context for all handlers of current logger.
+        """
+        for handler in self.logger.handlers:
+            try:
+                handler.set_context(task_instance)
+            except AttributeError:
```

Comment:
Same as above comment for `hasattr`. Standard Python handlers won't have set_context so the method can't exist for all methods. It's worth thinking that's the best way to set logger context without breaking the abstraction for python log handlers.